### ISSUE TITLE
Output files are several times larger than input

### ISSUE THREAD

**Issue by AtilioA** [WyattBlue/jumpcutterV2#9]
I don't recall how much of a problem this was with carykh's code but sometimes I'll get a file that is 3-8x larger as output, even if the video itself is shorter and both have the same quality. Maybe it just needs tweaking ffmpeg arguments a bit?

**Comment by WyattBlue** [comment]
@AtilioA  Try using the new version. I overhauled the code.

**Comment by AtilioA** [comment]
I've been testing it. Unfortunately, output size is the same and program runs ~1.5x slower.
Also, I think there's too much code in one file. What happened to splitDuration?
I've tried reintegrating some changes I made on the older version to this newer one (I saw you corrected some of my mistakes in the README) and I'll be opening PRs, I even got some .mkv files being correctly handled before but no luck right now.

It might be better to just rollback if performance doesn't get better.

**Comment by WyattBlue** [comment]
For file sizes, I think the reason why it's much bigger than the original is that it uses a different codec than the original. I don't think this is necessarily a bad thing since that means it's easer for editing programs/local media players to handle this new video file.

splitDuration was implemented to solve the problem that this program was taking way too much RAM. This implementation fixes that so there's no need to have this feature anymore.

I don't think 233 lines is too much for one file, considering the original jumpcutter was 203 lines long.

**Comment by AtilioA** [comment]
> splitDuration was implemented to solve the problem that this program was taking way too much RAM. This implementation fixes that so there's no need to have this feature anymore.
Fair enough. Is this why it is slower now? The last version wasn't taking much RAM.

**Comment by flyingzambie** [comment]
@AtilioA This version might be single threaded, the original method unintentionally had multi processing built in.

**Comment by AtilioA** [comment]
> @AtilioA This version might be single threaded, the original method unintentionally had multi processing built in.  I am currently trying to make a solution to split the files up and process each segment with another process to speed it up as an optional argument for people with a large amount of ram

That's something I was thinking about, even if it was already fast, I got 6 cores and 12 threads, so this takes at most 10-15% of CPU usage and very little of RAM.

> but learning how multi-processing works in python is taking a bit longer than expected lol.

Way easier than most languages at least haha

**Comment by flyingzambie** [comment]
This is something I slapped together real quick for testing but I think you might find this useful.

`import os
from subprocess import call
import multiprocessing
import tempfile
from shutil import rmtree

def splitFile(videoFile, filetype, TEMP):
  # splitting
  splitDuration = 1800
  splitVideo = 'ffmpeg -i "{}" -acodec copy -f segment -segment_time {} -vcodec copy -reset_timestamps 1 -map 0 {}/%d{}'.format(
      videoFile, splitDuration, TEMP, filetype
  )
  call(splitVideo, shell=True)

def combineFile(filename, TEMP):
  # mergeing
  generateFile = "for f in ./{}/*.mp4; do echo \"file '$f'\" >> mylist.txt; done".format(
      TEMP
  )
  call(generateFile, shell=True)

  concatVideo = ["ffmpeg", "-f", "concat", "-safe", "0", "-i", "mylist.txt"]
  concatVideo.extend(["-c", "copy", filename + "_faster.mp4"])

  call(concatVideo)

  os.remove("mylist.txt")

def multiProcessVideo(videoFile):
  filename, filetype = os.path.splitext(videoFile)
  TEMP = tempfile.mkdtemp()
  splitFile(videoFile, filetype, TEMP)

  # TODO add call for processing each segment here


  combineFile(filename, TEMP)


  rmtree(TEMP)`

**Comment by AtilioA** [comment]
I'd try using `multiprocessing`'s `Pool`:
`with Pool(n Threads) as p:
    p.map(process_chunk, chunks)`
This should assign a chunk to each thread. I think it wouldn't be synchronous but you should find `map_async` or something like that in the docs if needed. If the procedure isn't asynchronous we'd lost the original video's order, but I guess it would still be possible to retain it if you'd put something like the first frame of the chunk when writing them. I don't quite remember but the program might already do something similar.
